### ISSUE TITLE
crashes since macOS catalina

### ISSUE THREAD

**Issue by leost1** [SelfControlApp/selfcontrol#574]
the application crashes frequently since the new Catalina 10.15 system update and it seems to be most likely to happen after I turn off my computer and turn it on after.
 Thanks for all your involvement in the project and have a great day !

**Comment by paramsinghvc** [comment]
Yes, same issue here. I've updated to Catalina and since then it doesn't let me turn it on. All it shows is this small prompt and does nothing else. Please help.

<img width="271" alt="Screenshot 2019-10-14 at 08 36 12" src="https://user-images.githubusercontent.com/4329912/66727557-b22a5a00-ee5d-11e9-81b6-56c7a19cb29b.png">

**Comment by cstigler** [comment]
Hey folks, that is indeed annoying. If you're developers or otherwise tech-savvy, could you dig for some relevant logs that might help debug the issue? I've never used Catalina, so it's hard to know what's going on without logging.

Two ideas:
1. Open up Console.app and try to find crash logs starting with `SelfControl` or `org.eyebeam.SelfControl`. You can Google for "how to find crash logs mac" for more info if you're not sure.
2. Open up Console.app and go to `system.log`, and search for `selfcontrol` to see if there's any lines logged about us.

You can post logs here, or email to me at charlie@charliestigler.com. Thanks for the help!

**Comment by sharaquss** [comment]
I managed to get it up and running by clicking this `Stuck? Stop block manually` button. After providing password app goes back to normal (slider view).

**Comment by nguyenthanhvuh** [comment]
Same issue with Catalina.  The trick from @sharaquss helps.

**Comment by cstigler** [comment]
@sharaquss @nguyenthanhvuh so just to confirm, the issue was that the app was stuck on 00:00 when you first opened it on Catalina? And then once you used the "Stuck? Stop block manually" button, it has been working totally fine since then (i.e. no further issues)? Want to make sure I understand.

**Comment by nguyenthanhvuh** [comment]
@cstigler Yes for the first question.  But for the second,  it works after used the "Suck? Stop block manually". But when that session is done,  it goes back to being suck at 00:00 again (and I have to use the "Suck? Stop block manually method to fix it).

**Comment by cstigler** [comment]
@nguyenthanhvuh thanks for the info, one other question: when you start a block and then at the end it's stuck at 00:00, are your sites still being blocked? Or did the block actually come off, but for some reason the interface is still showing like it's running?

and if you have any interest in logging at your logs (see above), that would be awesome too!

**Comment by nguyenthanhvuh** [comment]
No, when the block session expires then the block comes off (as expected).  

No crash log but there are these lines in my `system.log`:

```
Oct 14 10:24:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 1 seconds. Pushing respawn out by 9 seconds.
Oct 14 12:22:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 7 seconds. Pushing respawn out by 3 seconds.
Oct 14 13:22:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 7 seconds. Pushing respawn out by 3 seconds.
Oct 14 14:22:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 7 seconds. Pushing respawn out by 3 seconds.
Oct 14 15:22:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 7 seconds. Pushing respawn out by 3 seconds.
Oct 14 16:22:00 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 7 seconds. Pushing respawn out by 3 seconds.
Oct 14 16:24:03 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 2 seconds. Pushing respawn out by 8 seconds.
Oct 14 17:24:04 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 4 seconds. Pushing respawn out by 6 seconds.
Oct 14 18:23:04 Saigon com.apple.xpc.launchd[1] (org.eyebeam.SelfControl): Service only ran for 4 seconds. Pushing respawn out by 6 seconds.
```

**Comment by cstigler** [comment]
Hmm, got it, thanks for looking. Unfortunately not sure why that's happening - but will take a look as soon as I get Catalina running on my machine.

**Comment by sbibauw** [comment]
I confirm the same behaviour as what @nguyenthanhvuh is describing (gets stuck at the end of the block, does not block the websites anymore after that). It is quite frustrating because it does not let me use [auto-selfcontrol](https://github.com/andreasgrill/auto-selfcontrol/) (automated daily scheduled run of SelfControl) anymore.

**Comment by nguyenthanhvuh** [comment]
@sbibauw thanks for pointing to`auto-selfcontro`l,  exactly what I have been looking for (but now of course have to wait for the fix to self-control itself).

**Comment by ricardo-sabatino** [comment]
Same thing happening here, used @sharaquss comment here to get it to work

**Comment by jarhill0** [comment]
Like @nguyenthanhvuh and @sbibauw I have the issue on Catalina where the block ends but the interface acts as if it's still running and I have to click the "stuck?" button every time.

**Comment by Ariaflame** [comment]
I note that when it reaches the end, which seems to display about 6 seconds to go, or at least 6 as the final digit is when it reaches the stuck condition. But the SelfControl-killer.log only updates in Documents once you've clicked on the bit to unstick the app, and put in your password so that it can have access. So is it an issue of where it's trying to update logs? Or timing for how long it has access for that?

**Comment by kmh1000** [comment]
My system running Catalina responds similarly. "Stuck? Stop block manually" doesn't work. The only thing I've found to work is resetting to defaults in Terminal with the command: defaults delete org.eyebeam.SelfControl

**Comment by Ariaflame** [comment]
Odd. I do get the Stuck? Stop block manually working fine, as soon as I put in my password to let it save the log file.

**Comment by cstigler** [comment]
Hi folks, I took a look at this issue on Catalina and think I've got it fixed. **Can you please download and test this beta version of SelfControl and let me know if it fixes your problems?** Steps:

1. Make sure that you do NOT have a SelfControl block active on your system, i.e. when you open SelfControl it shows the "Start" screen. If it's stuck on 00:00, please use the "Stuck? Stop block manually" button to get it back to the "Start" screen before proceeding.

2. Delete SelfControl from your Applications folder.

3. Download this new beta version of SelfControl, unzip it, and put it in your Applications folder: http://downloads.selfcontrolapp.com/SelfControl-2.3-beta.zip

4. Run a short block and make sure that the block ends as expected (note that it's normal for it to take up to 15-20 seconds in the "Finishing" state before it ends - give it a moment).

5. Test running different blocks, make sure that the app is working as you'd expect, make sure the UI is displaying in your preferred system language, etc. In short, make sure it's still working for you.

6. Report back in this thread how it went!

If all goes well here, I'll release this officially in the next few days.

**Comment by AshishKapoor** [comment]
<img width="258" alt="Screenshot 2019-10-22 at 5 03 19 PM" src="https://user-images.githubusercontent.com/5203107/67281805-e2bd6400-f4ed-11e9-987d-b3d9ef178b91.png">
Same! seeing this.

**Comment by nguyenthanhvuh** [comment]
Doesn't work for me.  I now get a screen that says:
```
Finishing
Add to Blacklist   -> changes to Stuck? Stop block manually after several secs.
Extend Block Timer 
```

**Comment by nguyenthanhvuh** [comment]
Not sure how I get the above screen.  But now after I use the Stop block manually and rerun the app,  it seems to work now.  Will try it with different time intervals and let you know.

**Comment by sbibauw** [comment]
It works for me! Tried two blocks (15min and 1hour), and afterwards, it returns to the normal ("Start", time slider, "Edit blacklist") screen. The only thing that seems to be missing is the localization (it was partial for French in previous versions, but now I don't see anything translated) (really not a problem for usability though, but just letting you know).

**Comment by ricardo-sabatino** [comment]
> Hi folks, I took a look at this issue on Catalina and think I've got it fixed. **Can you please download and test this beta version of SelfControl and let me know if it fixes your problems?**
> 
> Steps:
> 
> 1. Make sure that you do NOT have a SelfControl block active on your system, i.e. when you open SelfControl it shows the "Start" screen. If it's stuck on 00:00, please use the "Stuck? Stop block manually" button to get it back to the "Start" screen before proceeding.
> 2. Delete SelfControl from your Applications folder.
> 3. Download this new beta version of SelfControl, unzip it, and put it in your Applications folder: http://downloads.selfcontrolapp.com/SelfControl-2.3-beta.zip
> 4. Run a short block and make sure that the block ends as expected (note that it's normal for it to take up to 15-20 seconds in the "Finishing" state before it ends - give it a moment).
> 5. Test running different blocks, make sure that the app is working as you'd expect, make sure the UI is displaying in your preferred system language, etc. In short, make sure it's still working for you.
> 6. Report back in this thread how it went!
> 
> If all goes well here, I'll release this officially in the next few days.

Seems to be working fine for me! (Macbook Pro 16" 2018 - Catalina) one thing to keep in mind: the UI to add websites is hard to read in dark mode. Attached a screenshot below.

![image](https://user-images.githubusercontent.com/9910577/67304592-0e2a5980-f4c2-11e9-97dc-b1b41e40b99d.png)

**Comment by cstigler** [comment]
Thanks y'all for testing!

@nguyenthanhvuh: interesting, was it definitely on the "Start" screen before you upgraded (i.e. it wasn't already stuck running a block?). That would be expected behavior if it was already stuck before you upgraded...

@sbibauw: glad it's working! I think I see what caused the localization problems. If you download this updated beta version I think the French will be back for you: http://downloads.selfcontrolapp.com/SelfControl-2.3-beta2.zip

@AshishKapoor: if you'd like to try out a fix to your problem, please read the post above yours and try downloading the beta version of the application! it sounds like it'll probably fix your problem.

@ricardo-sabatino: glad it's working for you too! It does look pretty rough in that screenshot, but that'll be for another day

**Comment by ricardo-sabatino** [comment]
Thank you! This app keeps me employed 😂

**Comment by jarhill0** [comment]
Just another voice chiming in that the beta update seems to fix the problem. Thanks!

**Comment by AshishKapoor** [comment]
> > Hi folks, I took a look at this issue on Catalina and think I've got it fixed. **Can you please download and test this beta version of SelfControl and let me know if it fixes your problems?**
> > Steps:
> > 
> > 1. Make sure that you do NOT have a SelfControl block active on your system, i.e. when you open SelfControl it shows the "Start" screen. If it's stuck on 00:00, please use the "Stuck? Stop block manually" button to get it back to the "Start" screen before proceeding.
> > 2. Delete SelfControl from your Applications folder.
> > 3. Download this new beta version of SelfControl, unzip it, and put it in your Applications folder: http://downloads.selfcontrolapp.com/SelfControl-2.3-beta.zip
> > 4. Run a short block and make sure that the block ends as expected (note that it's normal for it to take up to 15-20 seconds in the "Finishing" state before it ends - give it a moment).
> > 5. Test running different blocks, make sure that the app is working as you'd expect, make sure the UI is displaying in your preferred system language, etc. In short, make sure it's still working for you.
> > 6. Report back in this thread how it went!
> > 
> > If all goes well here, I'll release this officially in the next few days.
> 
> Seems to be working fine for me! (Macbook Pro 16" 2018 - Catalina) one thing to keep in mind: the UI to add websites is hard to read in dark mode. Attached a screenshot below.
> 
> ![image](https://user-images.githubusercontent.com/9910577/67304592-0e2a5980-f4c2-11e9-97dc-b1b41e40b99d.png)

Not addicted to reddit yet! 🙈

**Comment by sbibauw** [comment]
> @sbibauw: glad it's working! I think I see what caused the localization problems. If you download this updated beta version I think the French will be back for you: http://downloads.selfcontrolapp.com/SelfControl-2.3-beta2.zip

Indeed! It works fine, and the localization is back!
Now going to reactivate my [auto-selfcontrol](https://github.com/andreasgrill/auto-selfcontrol/) workflow! 🎉

**Comment by kmh1000** [comment]
It seems to be working for me, too! Thanks for the fix! This is a great productivity program and I appreciate the effort.

**Comment by cstigler** [comment]
Thanks all for testing, glad it seems to be working all around!

I just released this as SelfControl 2.3. You can `Check for Updates` in your older SelfControl app and should see it pop up an an available update.

If you downloaded the beta version I linked above, it won't auto-update to the official release of 2.3. To upgrade manually:
1. download and unzip the official release of 2.3 from http://downloads.selfcontrolapp.com/SelfControl-2.3.zip
2. close SelfControl
3. replace your copy of SelfControl in `Applications` with the new one you just downloaded

Please let me know if you notice any regressions with the new release, or if it's not fixing your macOS Catalina problems!
